utils/registered_pools.go

```diff
@@ -0,0 +1,17 @@
+package utils
+
+type PoolParams struct {
+	from string
+	to   string
+	data string
+}
+
+// Get these data from individual pools
+var RegisteredPools = map[string]PoolParams{
```

Comment:
Have more than one pool here so we can see how it behaves with an array (as mentioned in previous comment, the string templating was missing a newline for example)